### PR TITLE
[Fluid] Acceleration limitation: exception for fixed velocity components

### PR DESCRIPTION
**Pull request by mrhashemi**
**Description**
Bugfix for fixed velocity.

Please mark the PR with appropriate tags: 
- Bugfix

**Changelog**
Please summarize the changes in one list to generate the changelog:
E.g.
- acceleration_limitation_utilities

### CODE REVIEW

**Review by rubenzorrilla — DISMISSED** [review]
Thanks!

File: applications/FluidDynamicsApplication/custom_utilities/acceleration_limitation_utilities.cpp [hunk 1]
```diff
@@ -34,31 +35,48 @@ namespace Kratos
...
-            ModelPart::NodesContainerType::iterator i = rNodes.begin() + count;
+        double max_delta_v = block_for_each<MaxReduction<double>>(mrModelPart.Nodes(), [&](Node<3>& rNode){
+            double norm_delta_v_fixed = 0.0;
+            if ( rNode.Is(INLET) || rNode.IsFixed(VELOCITY_X) || rNode.IsFixed(VELOCITY_Y) || rNode.IsFixed(VELOCITY_Z) ){
+                const array_1d<double, 3> &v  = rNode.FastGetSolutionStepValue( VELOCITY, 0 );
+                const array_1d<double, 3> &vn = rNode.FastGetSolutionStepValue( VELOCITY, 1 );
+                const array_1d<double, 3> delta_v = v - vn;
 
+                norm_delta_v_fixed = std::sqrt( delta_v[0]*delta_v[0] + delta_v[1]*delta_v[1] + delta_v[2]*delta_v[2] );
+            }
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/custom_utilities/acceleration_limitation_utilities.cpp]:
```suggestion
        array_1d<double,3> delta_v;
        double max_delta_v = block_for_each<MaxReduction<double>>(mrModelPart.Nodes(), delta_v, [&](Node<3>& rNode, array_1d<double,3>& rDeltaVTLS){
            double norm_delta_v_fixed = 0.0;
            if ( rNode.Is(INLET) || rNode.IsFixed(VELOCITY_X) || rNode.IsFixed(VELOCITY_Y) || rNode.IsFixed(VELOCITY_Z) ){
                const array_1d<double, 3> &r_v  = rNode.FastGetSolutionStepValue( VELOCITY, 0 );
                const array_1d<double, 3> &r_vn = rNode.FastGetSolutionStepValue( VELOCITY, 1 );
                rDeltaVTLS = r_v - r_vn;

                norm_delta_v_fixed = norm_2(rDeltaVTLS);
            }
```

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_utilities/acceleration_limitation_utilities.cpp]: This way we avoid allocating `delta_v` for each node.

**Review by rubenzorrilla — COMMENTED** [review]
Maybe I'm wrong, but with current changes we only consider the `INLET` and fix `VELOCITY` nodes for the acceleration limitation. Shouldn't we consider all the nodes in the domain to then apply the correction if the nodes are neither `INLET` nor fixed?

**Comment by mrhashemi** [comment]
> Maybe I'm wrong, but with current changes we only consider the `INLET` and fix `VELOCITY` nodes for the acceleration limitation. Shouldn't we consider all the nodes in the domain to then apply the correction if the nodes are neither `INLET` nor fixed?

max_delta_v is calculated for Fixed nodes and INLET, while the correction is done for not Fixed and not INLET nodes.

**Comment by rubenzorrilla** [comment]
> > Maybe I'm wrong, but with current changes we only consider the `INLET` and fix `VELOCITY` nodes for the acceleration limitation. Shouldn't we consider all the nodes in the domain to then apply the correction if the nodes are neither `INLET` nor fixed?
> 
> max_delta_v is calculated for Fixed nodes and INLET, while the correction is done for not Fixed and not INLET nodes.

That's my point. Shouldn't we calculate `max_delta_v` with all the domain nodes?

**Comment by mrhashemi** [comment]
> > > Maybe I'm wrong, but with current changes we only consider the `INLET` and fix `VELOCITY` nodes for the acceleration limitation. Shouldn't we consider all the nodes in the domain to then apply the correction if the nodes are neither `INLET` nor fixed?
> > 
> > 
> > max_delta_v is calculated for Fixed nodes and INLET, while the correction is done for not Fixed and not INLET nodes.
> 
> That's my point. Shouldn't we calculate `max_delta_v` with all the domain nodes?

This is an indicator of the maximum possible/allowed acceleration. In the second loop, the acceleration of all nodes is compared with this (as well as the gravitational) acceleration.

**Comment by rubenzorrilla** [comment]
> > > > Maybe I'm wrong, but with current changes we only consider the `INLET` and fix `VELOCITY` nodes for the acceleration limitation. Shouldn't we consider all the nodes in the domain to then apply the correction if the nodes are neither `INLET` nor fixed?
> > > 
> > > 
> > > max_delta_v is calculated for Fixed nodes and INLET, while the correction is done for not Fixed and not INLET nodes.
> > 
> > 
> > That's my point. Shouldn't we calculate `max_delta_v` with all the domain nodes?
> 
> This is an indicator of the maximum possible/allowed acceleration. In the second loop, the acceleration of all nodes is compared with this (as well as the gravitational) acceleration.

Sure. You're totally right. Forget about my observation. Approving.

**Review by rubenzorrilla — APPROVED** [review]

**Comment by philbucher** [comment]
I think this broke the nightly build, can you please take a look?

**Comment by mrhashemi** [comment]
> I think this broke the nightly build, can you please take a look?

@rubenzorrilla if you agree, maybe I can revert these changes until I can resolve the issue with the "nightly build".
As far as I can say there is only a call for NavierStokesTwoFluidsSolver in DarcyChannelTest. It should be the only instance of this AccLimitation in the tests.

**Comment by philbucher** [comment]
ping @rubenzorrilla

**Comment by rubenzorrilla** [comment]
In the nightly build log the test reported to fail is `testTwoFluidMassConservationTest3D`. The point seems to be a slight modification in the expected results by the changes in here so it should be pretty straightforward to fix. I wouldn't revert the changes in this case.

**Pull request merged**